docs/tutorial/codecs.txt

```diff
@@ -0,0 +1,108 @@
+======
+Codecs
+======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. versionadded:: 1.17
+
+Overview
+--------
+
+Codecs are used to decode BSON documents into PHP objects, and encode PHP objects into BSON documents. In contrast to
+other methods (e.g. type maps), codecs allow for greater customization and handling of different data types. They allow
+separating the logic for BSON encoding and decoding from the domain classes, allowing to decode BSON into plain old PHP
+objects (POPOs).
```

Comment:
Enjoy this gem I came across while googling for "POPOs": http://www.javaleaks.org/open-source/php/plain-old-php-object.html

Reply:
You'll have an army of lawyers suing you for making POPOs.

Reply:
Haha. Leaving this unresolved so that someone can eventually stumble upon this while cave diving into old pull requests.